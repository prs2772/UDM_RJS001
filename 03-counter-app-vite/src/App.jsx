import PropTypes from 'prop-types';

const message  = [1,2,3,4,5,6,7,8,,10];
const newMessage = {
  "eichOne": "H1",
  "eichTwo": "H2",
}

const App = ({ title, subTitle='', age=22 }) => {
  return(
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{message}</p>
      <p>{JSON.stringify(newMessage)}</p>
      <p>{`Title: ${title}`}</p>
      <p>{`${subTitle}`}</p>
      <p>{`${subTitle}`}</p>
      <p>{`Age: ${age}`}</p>
    </>
  );
}

App.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  age: PropTypes.number,
}
App.defaultPropTypes = {
  age: 23,
}

export { App };
