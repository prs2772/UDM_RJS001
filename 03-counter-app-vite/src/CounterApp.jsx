import {useState} from 'react'
import PropTypes from 'prop-types';

const CounterApp = ({value=0}) => {
  const buttonsType = ['counter-app-btnAdd', 'counter-app-btnSubstract', 'counter-app-btnReset'];
  const [counter, setCounter] = useState (value);
  //API... Re-render 
  console.log('render');

  const operation = (event, key) => {
    const caso = buttonsType.findIndex(e => e === key);
    setCounter(
      caso === 0 ? counter + 1 : 
        caso === 1 ? counter - 1 : value
    );
    console.log(event);
  }

  return(
    <div className='counter-app'>
      <h1>CounterApp</h1>
      <h2>{`Value: ${counter}`}</h2>
      <div className='counter-app__container-buttons--main'>
        <button key={buttonsType[0]} className='counter-app__button--add' onClick={e => operation(e, buttonsType[0])}>
          +1
        </button>
        <button key={buttonsType[1]} className='counter-app__button--substract' onClick={e => operation(e, buttonsType[1])}>
          -1
        </button>
        <button aria-label="counter-app__button--reset" key={buttonsType[2]} className='counter-app__button--reset' onClick={e => operation(e, buttonsType[2])}>
          Reset
        </button>
      </div>
    </div>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
}

export {CounterApp};
