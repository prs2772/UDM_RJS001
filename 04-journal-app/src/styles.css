/*https://coolors.co/7b7554-17183b-a11692-ff4f79-ffb49as*/
:root {
  --min-width: 320px;
}

html {
  font-size: 3px;
}

@media screen and (min-width: 480px) {
  html {
    font-size: 5px;
  }
}
@media screen and (min-width: 660px) {
  html {
    font-size: 7px;
  }
}
@media screen and (min-width: 900px) {
  html {
    font-size: 9px;
  }
}
* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: "Montserrat", sans-serif;
}

main {
  min-width: var(--min-width);
}

h1 {
  font-size: 3rem;
}

p, label, span {
  font-size: 1.5rem;
}

label {
  display: block;
  margin: 1rem 0;
}

section {
  margin: 3rem 0;
}

img {
  width: 100%;
}

.container-img {
  width: 80vw;
  margin: 3rem auto;
}

/*Overwrite*/
* {
  color: #FFFFFF;
  z-index: 0;
}

ul, ol {
  list-style: none;
}

html {
  background-color: #17183b;
}