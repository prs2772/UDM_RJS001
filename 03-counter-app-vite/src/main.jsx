import React from 'react';
import ReactDOM from 'react-dom/client';
import { App } from './App';
import { CounterApp } from './CounterApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/*<HelloWorldApp />*/}
    {/*<App username={'123'} description={'MIB Agent'} />*/}
    <CounterApp value={20}/>
  </React.StrictMode>
);
