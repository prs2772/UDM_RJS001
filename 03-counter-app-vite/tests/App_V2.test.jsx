import { render, screen } from "@testing-library/react";
import { App } from "../src/App";

describe("Pruebas en <App />", () => {
  const title = "Test App";
  const subTitle = "Subtitle";

  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<App title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar el mensaje "Test App"', () => {
    render(<App title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
    // screen.debug();
  });

  test("debe de mostrar el titulo en un h1", () => {
    render(<App title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("debe de mostrar el subtitulo enviado por props", () => {
    render(<App title={title} subTitle={subTitle} />);

    expect(screen.getAllByText(subTitle).length).toBe(2);
  });
});
