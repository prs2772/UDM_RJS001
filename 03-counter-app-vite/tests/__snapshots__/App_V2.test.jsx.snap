// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Test App
  </h1>
  <p>
    1
    2
    3
    4
    5
    6
    7
    8
    10
  </p>
  <p>
    {"eichOne":"H1","eichTwo":"H2"}
  </p>
  <p>
    Title: Test App
  </p>
  <p />
  <p />
  <p>
    Age: 22
  </p>
</div>
`;
